pkg/execution/history/lifecycle.go

```diff
@@ -482,6 +483,102 @@ func (l lifecycle) OnWaitForEventResumed(
 	}
 }
 
+// OnInvokeFunction is called when a function is invoked from a step.
+func (l lifecycle) OnInvokeFunction(
```

Comment:
change to `OnInvokeScheduled` to fit the rest of the naming params?

Reply:
Matching `OnWaitForEvent`, where we seem to have `On[Tool]`:
```diff
OnWaitForEvent
OnWaitForEventResumed
OnSleep
+ OnInvokeFunction
+ OnInvokeFunctionResumed
```